andela/ah-infinity-stones
# #163191975 Fix signup/login bugs

#### What does this PR do?
- Fix user registration/login bugs
#### Description of Task to be completed?
- Generate token on login
- Fix user account activation link
#### How should this be manually tested?
1. `git clone -b fix-authentication-bugs-163191975 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD=pass'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
export EMAIL_HOST_USER='username'
export EMAIL_HOST_PASSWORD='pass'
export SECRET_KEY='key-is-secret'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
{ "user": {
"username": "username44",
"email": "username@user.com",
"password": "@Password321"	
}
}
```
Check your email and activate account. Login with the following endpoint:
- `http://127.0.0.1:8000/api/users/login/` - Login 

8. Run tests`python3 manage.py test authors.apps.authentication.tests.test_registration`
#### What are the relevant pivotal tracker stories?
[#163191975](https://www.pivotaltracker.com/story/show/163191975)
#### Screenshots
<img width="948" alt="screenshot 2019-01-14 at 18 40 57" src="https://user-images.githubusercontent.com/12385482/51122907-fdaacb80-182b-11e9-8af3-df588e4cf337.png">
